packages/expo/src/environment/DevLoadingView.ios.tsx

```diff
@@ -0,0 +1,114 @@
+import React, { useEffect, useState, useRef } from 'react';
+import {
+  Animated,
+  StyleSheet,
+  SafeAreaView,
+  Text,
+  NativeModules,
+  NativeEventEmitter,
+  View,
+} from 'react-native';
+
+const NativeDevLoadingView = NativeModules.DevLoadingView;
+const nativeDevLoadingViewEventEmitter = new NativeEventEmitter(NativeDevLoadingView);
+
+export default function DevLoadingView() {
+  let [isDevLoading, setIsDevLoading] = useState(false);
+  let [isAnimating, setIsAnimating] = useState(false);
+  let translateY = useRef(new Animated.Value(0)).current;
+
+  useEffect(() => {
+    function handleShowMessage({ message }) {
+      // "Refreshing..." is the standard fast refresh message and it's the
+      // only time we want to display this overlay.
+      if (message !== 'Refreshing...') {
+        return;
+      }
+
+      // TODO: maybe we want to add a timeout here, so if it takes more than a
+      // couple seconds we ask the developer if they want to try a full reload?
+
+      setIsDevLoading(true);
+      translateY.setValue(0);
+    }
+
+    function handleHide() {
+      setIsAnimating(true);
+      setIsDevLoading(false);
+      Animated.timing(translateY, {
+        toValue: -100,
+        delay: 1000,
+        duration: 350,
+        useNativeDriver: true,
+      }).start(({ finished }) => {
+        if (finished) {
+          setIsAnimating(false);
+          translateY.setValue(0);  
+        }
+      });
+    }
+
+    nativeDevLoadingViewEventEmitter.addListener('devLoadingView:showMessage', handleShowMessage);
+    nativeDevLoadingViewEventEmitter.addListener('devLoadingView:hide', handleHide);
+
+    return function cleanup() {
+      nativeDevLoadingViewEventEmitter.removeListener(
+        'devLoadingView:showMessage',
+        handleShowMessage
+      );
+      nativeDevLoadingViewEventEmitter.removeListener('devLoadingView:hide', handleHide);
+    };
+  });
+
+  if (isDevLoading || isAnimating) {
+    return (
+      <Animated.View style={[styles.animatedContainer, { transform: [{ translateY }] }]}>
+        <SafeAreaView style={styles.banner}>
+          <View style={styles.contentContainer}>
+            <View style={{flexDirection: 'row'}}>
+              <Text style={styles.text}>
+                {isDevLoading ? 'Refreshing...' : 'Refreshed'}
+              </Text>
+            </View>
+            <View style={{flex: 1}}>
+              <Text style={styles.subtitle}>
+                {isDevLoading ? 'Using Fast Refresh' : "Don't see your changes? Reload the app"}
```

Comment:
Is it strange to show two messages that say the same thing? "Refreshing..." and "Using fast refresh" right on top of each other?

Reply:
I think it makes sense. To me, it reads as "Refreshing using Fast Refresh", and I assume that we intend to make sure the name of the feature 'fast refresh' is in there to make sure our users know that they're using it.

We could also try `'Fast Refreshing ...' : 'Fast Refreshed'` if we want to compress the refreshing state text and feature name into one line. With that change, we can keep the 'don't see your changes...' part as a static string below it.

Reply:
definitely open to using some other text here - or i suppose we could use no subtitle when refreshing. i kept put it there so the layout would be the same between refreshing/refreshed states and to be clear about what was causing the refresh